app/clients/sms/firetext.py

```diff
@@ -28,22 +51,26 @@ def init_app(self, config, *args, **kwargs):
     def get_name(self):
         return self.name
 
-    def send_sms(self, to, content):
+    def send_sms(self, to, content, reference):
 
         data = {
             "apiKey": self.api_key,
             "from": self.from_number,
             "to": to.replace('+', ''),
-            "message": content
+            "message": content,
+            "reference": reference
         }
 
+        start_time = monotonic()
         try:
-            start_time = monotonic()
             response = request(
                 "POST",
-                "https://www.firetext.co.uk/api/sendsms",
+                "https://www.firetext.co.uk/api/sendsms/json",
                 data=data
             )
+            firetext_response = response.json()
+            if firetext_response['code'] != 0:
+                raise FiretextClientException(firetext_response)
```

Comment:
Firetext exception to capture response. Often a status code is not used, so inspect the status code and raise if non-zero.